doc/man/rlineage.man
.TH Rotifer 0 rnexplorer
.SH NAME
.PP
rnexplorer \- Annotate, clean, reformat rneighbors output
.SH SYNOPSIS
.PP
\fBrnexplorer\fP [\fIOPTIONS\fP] [\fIFILE\fP]
.PP
# Simplest usage:
.PP
\fBrnexplorer\fP rneighbors.tsv \-of gi2operon > rneighbors.gi2operon
.SH DESCRIPTION
.PP
This program helps you explore snapshots of gene neighborhooods created by
.PP
rneighbors/acc2operon (Rotifer) and/or gi2operons (TASS)
.SH AUTHORS
.PP
Gilberto Kaihami; Robson Souza
.SH OPTIONS
.PP
This section describes program options, aliases and default values
.PP
\fBProgram options\fP
.PP
\fBfile\fP
.PP
Input table/gi2operon file. Pipe compatible
.PP
default: empty
.PP
\fB\-\-header\fP, \fB\-y\fP
.PP
Add more information to gi2operon header. The additional information must be a valid collumn
.PP
in the table format
.PP
example: rnexplorer \-y assembly \-y classification
.PP
default: empty
.PP
\fB\-\-outformat\fP, \fB\-of\fP
.PP
Set output format. Acceptable output formats are:
.PP
.RS
.nf
\- table:     all neighborhood data formatted as a text table
\- gi2operon: all neighborhood data formatted as gi2operon
.fi
.RE
.PP
default: same as input
.PP
\fB\-\-exclude_by_type\fP, \fB\-xt\fP
.PP
Exclude individual features type in the format KEY:VALUE.
.PP
The value could be an unique value (e.g. \-xt type:CDS) or a file containing one column with values in each line.
.PP
example: \-xt type:CDS
         \-xt seq\fItype:[FILE]
         \-xt seq\fPtype:plasmid
.PP
\fB\-\-include_by_type\fP, \fB\-it\fP
Include individual features type in the format KEY:VALUE.
.PP
The value could be an unique value (e.g. \-it type:CDS) or a file containing one column with values in each line.
.PP
example: \-it type:CDS
         \-it seq\fItype:[FILE]
         \-it seq\fPtype:plasmid
.PP
\fB\-\-annotation\fP, \fB\-a\fP
.PP
Add annotation column to the output.
.PP
Usage formats: [FILE], [COL\fIKEY:FILE], or [COL\fPKEY:FILE:COL_NAME]
.PP
The annotation file must contain at least two columns. The first column is the key and the second colum contains the annotation.
.PP
There are three ways to use this flag:
.PP
(1) Input just the annotation file it will map the first column against the pid column in the neighborhood table.
.PP
(2) You can pass the column to map and the file containing the annotation.
.PP
(3) You can pass 3 parameters, the first is the reference column in the neighborhood table, the second is the annotation file, and the last parameter is the new column name.
.PP
examples: (1) \-a FILE \[la]2\[ra] \-a pid:FILE \[la]2\[ra] \-a nucleotide:FILE \[la]3\[ra] \-a pid:[FILE]:annotation_col
.PP
default: empty
.PP
\fB\-\-padtable\fP, \fB\-p\fP
.PP
Format gi2operon output
.PP
default: empty
.PP
\fB\-\-exclude_column\fP, \fB\-xc\fP
.PP
Exclude a column in the output table
.PP
Select a columb by number (index starts from 0), column name or an interval
.PP
Usage formats: col_number, start..end, column name
.PP
examples: \-xc 0       (refers to the first column)
          \-xc 0..4    (refers from the first to the fifth column)
          \-xc product (refers to product column)
.PP
\fB\-\-ixclude_column\fP, \fB\-ic\fP
.PP
Include a column in the output table
.PP
Select a columb by number (index starts from 0), column name or an interval
.PP
Usage formats: col_number, start..end, column name
.PP
examples: \-ic 0       (refers to the first column)
          \-ic 0..4    (refers from the first to the fifth column)
          \-ic product (refers to product column)
.PP
\fB\-\-add_information\fP, \fB\-ai\fP
.PP
Include an existing column in the gi2operon output format
.PP
Usage: \-ai [Column_name]
.PP
example: \-ai classification
.SH Program options summary
.TS
allbox;
cb cb cb
r r c
r r c
r r c
r r c
r r c
r r c
r r c
r r c
r r c
r r c
.
Long name	Aliases	Type
\-\-header	\-y	list
\-\-outformat	\-of	string
\-\-exclude\fIby\fPtype	\-xt	list
\-\-include\fIby\fPtype	\-it	list
\-\-annotation	\-a	list
\-\-padtable	\-p	boolean
\-\-exclude_column	\-xc	list
\-\-include_column	\-it	list
\-\-add_information	\-ai	list
\-\-version		boolean
.TE


dev/old/rlineage.0d1.py
#!/usr/bin/env python3

import sys
sys.setrecursionlimit(10000)
sys.path.insert(0,'/home/kaihami/mymodules/')
from rotifer import taxonomy as taxon
import rotifer.neighborhood.neighborhood as nh
import pandas as pd
import numpy as np
import argparse
import rotifer.core.cli as corecli

__version__ = '0.1'
__authors__ = 'Gilberto Kaihami'

def parse_cli():
    parser = argparse.ArgumentParser(description = 'Return lineage from a NCBI lineage column')
    parser.add_argument('dataframe',

                        help = 'Input a table',
                        nargs = '*',
                        action = corecli.action.autoload, duplicates = False)

    parser.add_argument('--outformat', '-o', '--of', '-of', dest = 'outformat',
                        help = 'Output format (full, partial)',
                        default = 'full')

    parser.add_argument('--column', '-c', '--c',
                        dest = 'column',
                        default = 'classification',
                        help = 'Input column name containing NCBI lineage information (Default = classification)')

    parser.add_argument('-y', '--y',
                        '-header', '--header',
                        dest = 'header',
                        action = 'append',
                        help = 'Output columns header (default: Classification)',
                        default = ['all'],
                        type = str)

    parser.add_argument('-t', '--taxonomy',
                        '--t', '-taxonomy',
                        dest = 'taxonomy',
                        action = 'append',
                        help = 'Add taxonomic group in the output (Default: superkingdom, phylum, class) [Options: superkingdom, phylum, class, order, family, genus, species]',
                        default = []
                        )

    parser.add_argument('--version',
                        action = 'version',
                        version = corecli.version(program = 'rlineage',
                                                  version = __version__,
                                                  authors = __authors__,
                                                  description = 'Return lineage'))

    args = parser.parse_args()

    return args

if __name__ == '__main__':
    args = parse_cli()
    header = args.header

    if args.taxonomy:
        taxonomy = [x.lower() for x in args.taxonomy]
    else:
        taxonomy = ['superkingdom','phylum', 'class']

    df = nh.ipt2table(args.dataframe).df

#    print(header)
    df_header = list(df.columns)
    if len(header) == 1:
        header = df_header + ['lineage']
    else:
        header = [x for x in header if x != 'all']
#        print(header)

    df = df.reset_index()

    tax_df = taxon.taxonomy.df2tax(df).rlineage(taxonomy)
    tax_df = tax_df.reset_index()

    s2 = df.merge(tax_df, how = 'left', left_on = 'index',
                  right_on ='idx')

    s2 = s2[header]
    s2 = s2.fillna('.')
    s2 = s2.applymap(str)
    print('\t'.join(s2.columns))
    for i, row in s2.iterrows():
        print('\t'.join(row.values))



